agent/lib/kontena/service_pods/starter.rb

```diff
@@ -8,28 +9,28 @@ class Starter
       include Kontena::Logging
       include Common
 
-      attr_reader :service_id, :instance_number
+      attr_reader :service_pod, :hook_manager
 
-      # @param [String] service_id
-      # @param [Integer] instance_number
-      def initialize(service_id, instance_number)
-        @service_id = service_id
-        @instance_number = instance_number
+      # @param [ServicePod] service_pod
+      def initialize(service_pod)
+        @service_pod = service_pod
+        @hook_manager = Kontena::ServicePods::LifecycleHookManager.new(service_pod)
       end
 
       # @return [Docker::Container]
       def perform
-        service_container = get_container(self.service_id, self.instance_number)
+        service_container = get_container(service_pod.service_id, service_pod.instance_number)
         unless service_container.running?
           info "starting service: #{service_container.name_for_humans}"
           log_service_pod_event(
-            self.service_id, self.instance_number,
+            service_pod.service_id, service_pod.instance_number,
             "service:start_instance", "starting service instance #{service_container.name_for_humans}"
           )
-          service_container.restart!('timeout' => service_container.stop_grace_period)
+          hook_manager.on_pre_start(service_container)
```

Comment:
I personally disagree about running these hooks also during container restarts, but I can accept this as a design choice. 

BTW should this also then run the `post_start` hooks? It hasn't run those before. It will become more significant once the agent container restart policies are implemented, and all container restarts will start using `Kontena::ServicePods::Starter`. ATM docker restart policies will not run the start hooks.

Reply:
If we make `pre_start` hooks to be run on every restart, `post_start` should also be run similarly I think